### PR TITLE
applying override logic for new policy regarding not lawfully present applicants

### PR DESCRIPTION
**Pull request by erubinst**
https://www.pivotaltracker.com/n/projects/2640057/stories/185060664

- Adding new operation to apply override logic in ticket
- Adding a spec to handle new operation
- Calling the operation in determine_full_eligibility after Mitc determination is added to payload
- Adding spec to determine_full_eligibility_spec to test new operation call

FFs for this ticket:
ELIGIBILITY_OVERRIDE_IS_ENABLED covers the entire operation
MITC_OVERRIDE_NOT_LAWFULLY_PRESENT_PREGNANT_IS_ENABLED, MITC_OVERRIDE_NOT_LAWFULLY_PRESENT_CHIP_ELIGIBLE_IS_ENABLED, and MITC_OVERRIDE_NOT_LAWFULLY_PRESENT_UNDER_TWENTY_ONE_IS_ENABLED handle the specific overrides.  

These will be enabled in ME and disabled in DC once tested

### CODE REVIEW

File: spec/operations/eligibilities/determine_full_eligibility_spec.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 RSpec.describe ::Eligibilities::DetermineFullEligibility, dbclean: :after_each do
   before do
+    allow(MedicaidGatewayRegistry).to receive(:feature_enabled?).and_return(false)
```
Comment (mdkaraman) [inline spec/operations/eligibilities/determine_full_eligibility_spec.rb]: Is there a reason to mock all feature flags as disabled?  May want to consider using `.and_call_original` in place of `.and_return(false)` if this is just to placate the spec.

File: app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        age = age(thhm)
+        nineteen_to_twenty_one = age > 18 && age < 21
+        nineteen_to_twenty_one && MedicaidGatewayRegistry[:eligibility_override].settings(:mitc_override_not_lawfully_present_under_twenty_one).item
+      end
+
+      def under_eighteen_chip_override?(thhm)
+        age(thhm) < 19 && MedicaidGatewayRegistry[:eligibility_override].settings(:mitc_override_not_lawfully_present_chip_eligible).item
+      end
+
+      def age(thhm)
```
Comment (mdkaraman) [inline app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb]: Calculating age dynamically could be problematic in some edge cases.  We should use the age stored on the application at the time of submission in EA.  There is a field in the applicant hash called `age_of_applicant` that can be used for this.

**Review by scaustin34 — APPROVED** [review]
Looks good to me!

**Review by mdkaraman — APPROVED** [review]
Adds a new operation to apply medicaid/CHIP eligibility overrides and inserts a call to this new operation into the full eligibility determination flow.  Nice job on the thorough spec coverage!

**Pull request merged**